src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -340,6 +345,15 @@ static byte GetCertificateVersion(X509Certificate2 c)
                 result = EnsureCertificateResult.FailedToTrustTheCertificate;
                 return result;
             }
+
+            if (result == EnsureCertificateResult.ValidCertificatePresent)
+            {
+                result = EnsureCertificateResult.ExistingHttpsCertificateTrusted;
+            }
+            else
+            {
+                result = EnsureCertificateResult.NewHttpsCertificateTrusted;
+            }
```

Comment:
Can we avoid adding new results here? This is a contract between us and the tooling teams, so adding new results changes that contract and might break them.

Reply:
@javiercn Ah, I see, good to know. I'll check out how the tooling stuff consumes this today.

I'll also add a note mentioning this contract next to the enum declaration so someone doesn't accidentally break the contract in the future.

Reply:
Hmm... can you point me to where this is used by tooling? I can't seem to find it.

cc @vijayrkn

Reply:
I am assuming you are referring to this: https://devdiv.visualstudio.com/DevDiv/_git/WebTools?path=/src/ProjectSystem/Web/CommonServices/AspNetCoreDeveloperCertificatesTool.cs&_a=contents&version=GBmain

Reply:
Thanks @vijayrkn! Looks like the tooling just calls the dev-certs tool and relies on some of the status codes here which are the exit codes of the tool: https://github.com/dotnet/aspnetcore/blob/36e14568e7cfcb12bcf7bcc04f7a7bfb7c071030/src/Tools/dotnet-dev-certs/src/Program.cs#L19-L26

The `EnsureCertificateResult` enum changes I made won't change the actual exit codes from the tool, since these turn into "Success" here:

https://github.com/dotnet/aspnetcore/pull/42251/files#diff-1454ec4889b7f660676a246d1b0f74f29b980e4370e1932e012e2b968d33702aR433

So I think we're safe.

Reply:
Adding @BillHiebert in case there are other places where we call into the tool.

Reply:
Added a note of caution about modifying the exit codes.

Reply:
This also impacts VS 4 Mac and Docker container tools. FYI.